io/private/EptSupport.hpp

```diff
@@ -90,6 +158,127 @@ inline std::array<double, 3> toArray3(const Json::Value& json)
     return result;
 }
 
+class PDAL_DLL Key
+{
+    // An EPT key representation (see https://git.io/fAiBh).  A depth/X/Y/Z key
+    // representing a data node, as well as the bounds of the contained data.
+public:
+    Key(BOX3D b) : b(b) { }
```

Comment:
I find this structure confusing.  Why don't you initialize b when you set x/y/z/d and vice versa?  Maybe I just find the [] operator to get at b kind of opaque.

Also, this code seems really familiar.  Is this a copy from somewhere?

Reply:
Remove bounds from constructor at 1629a3d.  Not really a copy, but the `operator[]` is used within Entwine too since it's often convenient to treat coordinates for points and bounds numerically rather than via names.